rust/lancedb/src/remote/table.rs

```diff
@@ -41,20 +44,47 @@ impl<S: HttpSend> RemoteTable<S> {
         let request = self.client.post(&format!("/table/{}/describe/", self.name));
         let response = self.client.send(request).await?;
 
-        if response.status() == StatusCode::NOT_FOUND {
-            return Err(Error::TableNotFound {
-                name: self.name.clone(),
-            });
-        }
-
-        let response = self.client.check_response(response).await?;
+        let response = self.check_table_response(response).await?;
 
         let body = response.text().await?;
 
         serde_json::from_str(&body).map_err(|e| Error::Http {
             message: format!("Failed to parse table description: {}", e),
         })
     }
+
+    fn reader_as_body(data: Box<dyn RecordBatchReader + Send>) -> Result<reqwest::Body> {
+        // TODO: Once Phalanx supports compression, we should use it here.
+        let mut writer = arrow_ipc::writer::StreamWriter::try_new(Vec::new(), &data.schema())?;
+
+        //  Mutex is just here to make it sync. We shouldn't have any contention.
+        let mut data = Mutex::new(data);
+        let body_iter = std::iter::from_fn(move || match data.get_mut().unwrap().next() {
+            Some(Ok(batch)) => {
+                writer.write(&batch).ok()?;
+                let buffer = std::mem::take(writer.get_mut());
+                Some(Ok(buffer))
+            }
+            Some(Err(e)) => Some(Err(e)),
+            None => {
+                writer.finish().ok()?;
+                let buffer = std::mem::take(writer.get_mut());
+                Some(Ok(buffer))
+            }
+        });
+        let body_stream = futures::stream::iter(body_iter);
```

Comment:
Very cool, pity no async writer but I suppose we could wrap in `spawn_blocking` calls at some point in the future.